tedana/resources/decision_trees/demo_minimal_external_regressors_motion_task_models.json

```
@@ -0,0 +1,271 @@
+{
+    "tree_id": "demo_minimal_external_regressors_motion_task_models",
+    "info": "Demonstration based on the minimal decision tree that uses partial F stats on a model with multiple external regressors divided by category and task regressors to bias towards keeping.",
+    "report": "This is based on the minimal criteria of the original MEICA decision tree \\citep{kundu2013integrated} without the more aggressive noise removal steps \\citep{dupre2021te}.",
+    "necessary_metrics": [
+        "kappa",
+        "rho",
+        "countsigFS0",
+        "countsigFT2",
+        "dice_FS0",
+        "dice_FT2",
+        "signal-noise_t",
+        "variance explained",
+        "pval Full Model",
+        "R2stat Full Model"
+    ],
+    "intermediate_classifications": ["provisionalaccept", "provisionalreject"],
+    "classification_tags": ["Likely BOLD", "Unlikely BOLD", "Low variance"],
+    "external_regressor_config": {
+        "regress_ID": "Fmodel",
+        "info": "Fits all external regressors to a single model using an F statistic",
+        "report": "Unspecified external nuisance regressors that were fit to components using a linear model were rejected if they fit.",
+        "detrend": true,
+        "calc_stats": "F",
+        "f_stats_partial_models": ["Motion", "CSF"],
+        "Motion": ["^mot_.*$"],
+        "CSF": ["^csf.*$"],
+        "task_keep": ["^signal.*$"]
```

Our meeting today got me thinking that I'd rather see all models specifically defined, rather than using a full model and a special field (`task_keep`).

Something like this:

```suggestion
        "f_stats_partial_models": ["Full", "Motion", "CSF", "Task"],
        "Full": ["(?!signal).*$"],
        "Motion": ["^mot_.*$"],
        "CSF": ["^csf.*$"],
        "Task": ["^signal.*$"]
```

---

Also, instead of having separate lookup and definitions, what about having the models in a dictionary?

```suggestion
        "models": {
            "Full": ["^(?!signal).*$"],
            "Motion": ["^mot_.*$"],
            "CSF": ["^csf.*$"],
            "Task": ["^signal.*$"]
        },
```

---

I think we'd still need special key words because this might obscure what actually happening with the partial models.
`Full` is all nuissance regressors. and the F test for `Full` is all nuissance regressors compared to the baseline detrending regressors.
The partial F statistics are the Full model compared to the Full model **excluding** the partial regressors of interest. That is the partial F statistic for motion uses all regressors except the motion regressors and the stat is effectively testing if the full model with the motion regressors models the data significantly better than the full model excluding motion regressors.
With the ways you've set this up `Full` would still need to be a special key work to define the regressors in the full model.
Similary, `task_keep` is not a partial F stat compared to the full model. It's a completely separate F stat, so just including it in a dictionary of models would lose that distinction.
I'm fine changing to a `models` dictionary, but we'd still need to predefine `Full` and `Task` as special keywords.

We also previously discussed letting users specify which regressors are part of the full model. You said that's an unncessary feature to add. I'm happy to add if you now think it would be useful.

---

>The partial F statistics are the Full model compared to the Full model excluding the partial regressors of interest. That is the partial F statistic for motion uses all regressors except the motion regressors and the stat is effectively testing if the full model with the motion regressors models the data significantly better than the full model excluding motion regressors.

I forgot that. Thanks for clarifying. Is there any particular benefit to evaluating partial F-stats instead of running full models for each of the sets of regressors?

>We also previously discussed letting users specify which regressors are part of the full model. You said that's an unncessary feature to add. I'm happy to add if you now think it would be useful.

If we're running separate full models for different sets of regressors (which is essentially what we're doing with nuisance vs. task regressors), it does seem like we should accept patterns for the different full models. I'm sorry for going back on what I originally thought on this.

I don't want to hold this PR up any further, so we can discuss this in a future PR or release, but what about something like this? 

```json
"external_regressor_config": [
    {
        "regress_ID": "F_nuisance",
        "info": "Fits all non-task regressors to a single model using an F statistic",
        "report": "Unspecified external nuisance regressors that were fit to components using a linear model were rejected if they fit.",
        "detrend": true,
        "statistics": ["F", "p", "R2"],
        "regressors": ["^(?!signal).*$"],
        "partial_models": {
            "Motion": ["^mot_.*$"],
            "CSF": ["^csf.*$"]
        }
    },
    {
        "regress_ID": "F_task",
        "info": "Fits all task regressors to a single model using an F statistic",
        "report": "External task regressors that were fit to components using a linear model were accepted if they fit.",
        "detrend": true,
        "statistics": ["F", "p", "R2"],
        "regressors": ["^signal.*$"]
    }
]
```

---

There are several benefits of partial models. (In the docstring I'm citing https://afni.nimh.nih.gov/pub/dist/doc/manual/3dDeconvolve.pdf which has more of an explanation of page 14). For our needs, the key benefit is that we get a full fit of all nuisance regressors so that, if a combination is significantly fit with a combination of motion and respiration regressors (but not either separately) it will be identified. Also, dependence between different groups of regressors in accounted for in a single model. That is, if head motion and a CSF ROI both contain some of the same effects, the total variance explained is calculated in a single model rather than calculated twice in 2 separate models. This is an oversimplifications, but if Motion R^2=0.3 and a separate CSF R^2=0.3, that might imply these account for a combined variance of R^2=0.6 (oversimplification!), but the R^2 for a combined model would only be 0.4.

The other general benefit of partial models is that it's a single statistical test, which is mathematically more principled. That said, we're not using these for rigorous statistical thresholds, so I doubt this actually matters to us.

I do like your proposed redesign of the configuration. I think it won't be too much work, it will make the code and documentation cleaner, and will give users more flexibility in submitting external regressor files. It's worth doing. That said, to print before I travel, I need to have a finished OHBM poster by Monday, so there's no way I'm doing it before then. Even without this change, I'm doubtful we'll get this PR merged by Monday, so I'd rather get this right than set an artificial deadline that rushes review of a non-trivial code change.

For your proposed specification, the two things I'd change are:
- `"statistics"` to `"statistic"` and just include `"F"` Selecting `F` will calculate `p` and `R2` based on the F stats. Particularly since this is our only option now, adding flexibility for a range of output options just adds complexity.
- Change the `regress_ID`s to `"nuissance"` and `"task"`. That would make it slightly easier to then have the metrics outputs be `nuissance F` `nuissance R2` `nuissance p` without the repeated `F_`. at the beginning of each one.

---

@tsalo I still need to update the RTD documentation, but this change is now fully implemented.